### PR TITLE
Update msal4j and msal4j extension version

### LINKED ISSUE #13201: Upgrade azure-identity msal4j from 1.3 to latest
Tracking work item to investigate upgrading from the Jan 2020 (1.3.0) release of msal4j to a later release, which now has additional bug fixes and better support for throttling instructions

https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/dev/changelog.txt

### PR DESCRIPTION
**Pull request by jianghaolu**
fixes #13201

### CODE REVIEW

**Comment by g2vinay** [comment]
@jianghaolu  Spring code needs to be updated to be compatible with latest msal.

**Comment by saragluna** [comment]
@jianghaolu not sure about the correct process, but should the Spring code fix by our side or？

**Comment by jianghaolu** [comment]
@saragluna spring module depends on identity - and we have a rule in the repo that modules should depend on the latest track 2 library.
But for spring modules to depend on latest identity with this change, you will also need to bump msal4j, otherwise they will be incompatible.
The PR has a second part here: https://github.com/Azure/azure-sdk-for-java/pull/13216 and has been tested and reviewed by @yiliuTo . Please let me know if you want us to hold this for your further testing.

File: sdk/identity/azure-identity/README.md [hunk 1]
```diff
@@ -11,32 +11,8 @@ The Azure Identity library provides Azure Active Directory token authentication
 
   [Source code][source] | [API reference documentation][javadoc] | [Azure Active Directory documentation][aad_doc]
 
-## Table of contents
```
Comment (mitchdenny) [inline sdk/identity/azure-identity/README.md:14]: Why is this removed? Not required anymore, doesn't conform to README standards?

Comment (jianghaolu) [reply sdk/identity/azure-identity/README.md:14]: This was removed based on arch board feedback

**Comment by jianghaolu** [comment]
@alzimmermsft It should be a minor version bump. But considering the confusion it will cause after shipping this minor change as a "1.1.0" after lots of new features in beta.1 through beta.6, and that the actual 1.1.0 is coming in a couple weeks, I'm hoping we can get away with this.
